hilti/toolchain/src/compiler/codegen/types.cc

```diff
@@ -1111,6 +1111,26 @@ const CxxTypeInfo& CodeGen::_getOrCreateTypeInfo(const hilti::Type& t) {
         });
 }
 
+cxx::Expression CodeGen::_applyLhs(cxx::Expression expr, const Type& type, bool lhs) {
```

Comment:
WDYT about moving `lhs` into the caller so that this function becomes a simple conversion to a LHS? We'd need the same boilerplate in the callers, but this would still feel cleaner to me, e.g.,

```cpp
    if ( auto x = Visitor(this).dispatch(c) ) {
        if ( ! lhs )
            return *x;
        else
            return _applyLhs(*x, c.type());
    }
```